src/components/my-widgets-scores.jsx

```diff
@@ -1,60 +1,84 @@
 	const [state, setState] = useState({
 		scores: [],
 		isShowingAll: false,
-		isLoadingScores: true
+		isLoadingScores: true,
+		hasScores: false,
+		showExport: false
 	})
 
-	useEffect(
-		() => {
-			setState({...state, scores: [], isShowingAll: false,  isLoadingScores: true})
-			// getScores
-			const options = {
-				"headers": {
-				"cache-control": "no-cache",
-				"pragma": "no-cache",
-				"content-type": "application/x-www-form-urlencoded; charset=UTF-8"
-				},
-				"body": `data=%5B%22${inst.id}%22%2C${true}%5D`,
-				"method": "POST",
-				"mode": "cors",
-				"credentials": "include"
-			}
+	// Initializes the data when widget changes
+	useEffect(() => {
+		setState({...state, scores: [], isShowingAll: false,  isLoadingScores: true})
+		// getScores
+		const options = {
+			"headers": {
+			"cache-control": "no-cache",
+			"pragma": "no-cache",
+			"content-type": "application/x-www-form-urlencoded; charset=UTF-8"
+			},
+			"body": `data=%5B%22${inst.id}%22%2C${true}%5D`,
+			"method": "POST",
+			"mode": "cors",
+			"credentials": "include"
+		}
 
-			fetch('/api/json/score_summary_get/', options)
-				.then(resp => {
-					if(resp.ok && resp.status !== 204) return resp.json()
-					return []
-				})
-				.then(scores => {
-					const ranges = [
-						"0-9",
-						"10-19",
-						"20-29",
-						"30-39",
-						"40-49",
-						"50-59",
-						"60-69",
-						"70-79",
-						"80-89",
-						"90-100",
-					]
-					scores.forEach(semester => {
-						semester.graphData = semester.distribution.map((d, i) => ({ label: ranges[i], value: d }) )
-						semester.totalScores = semester.distribution.reduce((total, count) => total+count)
-					})
+		fetch('/api/json/score_summary_get/', options)
+		.then(resp => {
+			if(resp.ok && resp.status !== 204 && resp.status !== 502) return resp.json()
+			return []
+		})
+		.then(scores => {
+			let _hasScores = false
+			const ranges = [
+				"0-9",
+				"10-19",
+				"20-29",
+				"30-39",
+				"40-49",
+				"50-59",
+				"60-69",
+				"70-79",
+				"80-89",
+				"90-100",
+			]
+			scores.forEach(semester => {
+				semester.graphData = semester.distribution?.map((d, i) => ({ label: ranges[i], value: d }) )
+				semester.totalScores = semester.distribution?.reduce((total, count) => total+count)
 
-					setState({
-						scores,
-						isShowingAll: scores.length < 2,
-						isLoadingScores: false
-					})
-				})
-		}, [inst.id]
-	)
+				if (semester.distribution !== undefined) {
+					_hasScores = true
+				}
+			})
+
+			setState({
+				scores,
+				isShowingAll: scores.length < 2,
+				isLoadingScores: false,
+				hasScores: _hasScores,
+				showExport: false
+			})
+		})
+		.catch(error => {
+			//console.log(error)
+		})
+	}, [inst.id])
+
+	const openExport = () => {
+		setState({...state, showExport: true})
+		document.body.style.overflow = "hidden"
+	}
+
+	const closeExport = () => {
+		setState({...state, showExport: false})
+		document.body.style.overflow = "auto"
+	}
```

Comment:
seeing a `document.something = something` set's off my "side effect" warning.  this should probably be a useEffect

```suggestion
	useEffect(() => {
		document.body.style.overflow = state.showExport ? 'hidden' : 'auto'
	}, [state.showExport]) 
	
	const openExport = () => {
		setState({...state, showExport: true})
	}

	const closeExport = () => {
		setState({...state, showExport: false})
	}
```